### ISSUE TITLE
Can't run Chrome 127 - Error: WebSocket connection closed

### ISSUE THREAD

**Issue by sam-yh** [DevExpress/testcafe#8240]
### What is your Scenario?

Since the update of Chrome to v.127, it can't be used with Testcafe anymore. Trying leads to the following error message:

```
ERROR Unable to open the "chrome:" browser due to the following error:
Error: WebSocket connection closed
    at /home/$USER/test_automation/node_modules/chrome-remote-interface/lib/chrome.js:94:35
    at Chrome._handleConnectionClose (/home/$USER/test_automation/node_modules/chrome-remote-interface/lib/chrome.js:256:13)
    at WebSocket.<anonymous> (/home/$USER/test_automation/node_modules/chrome-remote-interface/lib/chrome.js:243:22)
    at WebSocket.emit (node:events:519:28)
    at WebSocket.emit (node:domain:488:12)
    at WebSocket.emitClose (/home/$USER/test_automation/node_modules/ws/lib/websocket.js:246:10)
    at Socket.socketOnClose (/home/$USER/test_automation/node_modules/ws/lib/websocket.js:1148:15)
    at Socket.emit (node:events:519:28)
    at Socket.emit (node:domain:488:12)
    at TCP.<anonymous> (node:net:339:12)

Type "testcafe -h" for help.
```

### What is the Current behavior?

Can't run Testcafe test files with Chrome version 127

### What is the Expected behavior?

Being able to run Testcafe test files with Chrome version 127

### What is the public URL of the test page? (attach your complete example)

https://testcafe.io/

### What is your TestCafe test code?

```
import { Selector } from 'testcafe'

fixture`Bug Repro`.page`${'https://testcafe.io'}`

test('Bug Repro', async (t) => {
  await t.expect(Selector('.logo').exists).ok()
})

```

### Steps to Reproduce

1. Have Chrome installed with version 127
2. Run a test file with the given test code using Chrome 127

### TestCafe version

3.6.2.

### Node.js version

20.16.0

### Command-line arguments

`testcafe chrome test.js`

### Browser name(s) and version(s)

Chrome 127.0.6533.72

### Platform(s) and version(s)

Ubuntu 22.04.3 LTS

**Comment by sam-yh** [comment]
Note that using `--disable-native-automation` does circumvent the WebSocket error. Though at least for us it doesn't really seem to be a true workaround, since with it we run into constant time-outs.
Also, not sure if in any way related, in that mode, Chrome seems to then always display a dialog for choosing a search engine.

**Comment by enriccarrion** [comment]
Hey @sam-yh , yes it's related to the search engine selection screen.
You can use `--disable-search-engine-choice-screen` to avoid it, it worked in my case and the `WebSocket connection closed` is gone.

**Comment by sam-yh** [comment]
Hi @enriccarrion - thanks! Though that doesn't seem to be an option that exists for Testcafe. At least I couldn't find any mention in the documentation or recent commits and it also unfortunately doesn't seem to have any effect for us.

**Comment by gforepsly** [comment]
We are facing the same issue.

Running chrome with the next command works:
**`chrome --disable-search-engine-choice-screen`**

This is a chrome flag, but it works if provided with the browser. 

Although this workaround works, it's not something we can use on our side. Will there be some kind of a fix?

**Comment by sam-yh** [comment]
Just in case anyone else has issues applying the workaround, I had to use it this way, for it to work:
`testcafe chrome:--disable-search-engine-choice-screen PATH_TO_TEST`

**Comment by enriccarrion** [comment]
Yes @sam-yh it's a chrome.exe flag, not a TestCafe flag.

**Comment by orcwarrior** [comment]
> Just in case anyone else has issues applying the workaround, I had to use it this way, for it to work: `testcafe chrome:--disable-search-engine-choice-screen PATH_TO_TEST`

Confirmed, I believe its correct order for testcafe args

**Comment by johndwz** [comment]
I had to apply it to the runner: 
..
.browsers('chrome --window-size=1280,1024 --disable-search-engine-choice-screen')

Problem fixed for now.

**Comment by vdedhia** [comment]
Meanwhile how do I run it in headless mode ?
Or for a long term solution how does the `.testcaferc.json` file be updated ?

Coz when I am running through the command `testcafe chrome:--disable-search-engine-choice-screen` it is starting chrome instances. I have tried passing multiple parameters but to no avail.

**Comment by vdedhia** [comment]
> > Meanwhile how do I run it in headless mode ? Or for a long term solution how does the `.testcaferc.json` file be updated ?
> > Coz when I am running through the command `testcafe chrome:--disable-search-engine-choice-screen` it is starting chrome instances. I have tried passing multiple parameters but to no avail.
> 
> Hey, for an individual command, you need to use `" "`, e.g. `testcafe "chrome: --disable-search-engine-choice-screen" path/to/file`

Thanks for the reply, but this doesn't work as expected. It still starts chrome instances.
Command I used: `testcafe "chrome: --disable-search-engine-choice-screen chrome:headless" PATH_TO_FILE`

**Comment by sam-yh** [comment]
> > > Meanwhile how do I run it in headless mode ? Or for a long term solution how does the `.testcaferc.json` file be updated ?
> > > Coz when I am running through the command `testcafe chrome:--disable-search-engine-choice-screen` it is starting chrome instances. I have tried passing multiple parameters but to no avail.
> > 
> > 
> > Hey, for an individual command, you need to use `" "`, e.g. `testcafe "chrome: --disable-search-engine-choice-screen" path/to/file`
> 
> Thanks for the reply, but this doesn't work as expected. It still starts chrome instances. Command I used: `testcafe "chrome: --disable-search-engine-choice-screen chrome:headless" PATH_TO_FILE`

Ah sorry, I misread. I use `testcafe chrome:headless --disable-search-engine-choice-screen path/to/file` which works for me. The `" "` might get necesary if you do **not** use headless or use multiple chrome options.

**Comment by vdedhia** [comment]
Thanks, that worked like a charm 👍🏼

**Comment by gforepsly** [comment]
As of Chrome 130.0.6723.59 using Node 20.17.0 the websocket closed error started to occur almost every time, again. When it starts, it doesn't even want to maximizeWindow using beforeEach blocks (which was working up until now).

Anyone else experiencing the same issue?

`Error: WebSocket connection closed
    at /node_modules/chrome-remote-interface/lib/chrome.js:94:35
    at Chrome._handleConnectionClose (/node_modules/chrome-remote-interface/lib/chrome.js:256:13)
    at WebSocket.<anonymous> (/node_modules/chrome-remote-interface/lib/chrome.js:243:22)
    at WebSocket.emit (node:events:519:28)
    at WebSocket.emit (node:domain:488:12)
    at WebSocket.emitClose (/node_modules/chrome-remote-interface/node_modules/ws/lib/websocket.js:246:10)
    at Socket.socketOnClose (/node_modules/chrome-remote-interface/node_modules/ws/lib/websocket.js:1150:15)
    at Socket.emit (node:events:519:28)
    at Socket.emit (node:domain:488:12)
    at TCP.<anonymous> (node:net:339:12)`

**Comment by Bayheck** [comment]
Hello,

I believe you have encountered the following issue: [https://github.com/DevExpress/testcafe/issues/8286](https://github.com/DevExpress/testcafe/issues/8286)

We have released testcafe 3.7.0-rc.1 to address this issue.

 

Please take a look and let us know your results.

I would also like to ask you to create a separate issue ticket should you find a new issue by following this template:

[https://github.com/DevExpress/testcafe/issues/new?assignees=&labels=TYPE%3A+bug&template=bug_report.yaml](https://github.com/DevExpress/testcafe/issues/new?assignees=&labels=TYPE%3A+bug&template=bug_report.yaml)

**Comment by sirtawast** [comment]
We've switched to `chromium` and it seems to work 1:1 like chrome. RC1 does work without throwing `Error: WebSocket connection closed` but it fails to start the tests because of "What's new" tab.
